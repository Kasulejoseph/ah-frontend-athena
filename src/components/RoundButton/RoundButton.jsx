import React from "react";
import PropTypes from "prop-types";
import "./RoundButton.scss";
import { MDBIcon } from "mdbreact";

export default function RoundButton(props) {
  const { background } = props;
  const { onClick } = props;
  const { label } = props;
  const { hoverClass } = props;
  const { icon } = props;
  const classes = `my-round bg-${background} border relative d-flex flex-row m-1 p-2 align-items-center ${hoverClass}`;
  return (
    <div onClick={onClick} className={classes} role="presentation">
      <MDBIcon icon={icon} className="mr-2" />
      <p className="m-0">{label}</p>
    </div>
  );
}

RoundButton.propTypes = {
  onClick: PropTypes.func,
  background: PropTypes.string,
  label: PropTypes.string,
  icon: PropTypes.string,
  hoverClass: PropTypes.string
};

RoundButton.defaultProps = {
  onClick: () => {},
  background: "",
  label: "",
  icon: "",
  hoverClass: ""
};
