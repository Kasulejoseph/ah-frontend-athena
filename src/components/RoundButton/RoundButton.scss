.my-round {
  border-radius: 20px;
}

.hover {
  &:hover {
    border: 1px solid red !important;
    color: red;
  }
}
