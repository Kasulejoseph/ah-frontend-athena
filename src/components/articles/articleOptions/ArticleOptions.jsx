import React from "react";
import PropTypes from "prop-types";
import "./ArticleOptions.scss";
import RoundButton from "../../RoundButton/RoundButton";

const ArticleOptions = props => {
  const { canModify } = props;
  const { handleDelete } = props;
  const { handleLink } = props;

  if (canModify) {
    return (
      <div className="d-flex flex-colomn">
        <RoundButton
          icon="edit"
          label="Edit Article"
          hoverClass="hover"
          onClick={handleLink}
        />
        <RoundButton
          icon="trash"
          hoverClass="hover"
          label="Delete Article"
          onClick={handleDelete}
        />
      </div>
    );
  }

  return <span />;
};

ArticleOptions.propTypes = {
  handleDelete: PropTypes.func,
  handleLink: PropTypes.func,
  canModify: PropTypes.bool
};

ArticleOptions.defaultProps = {
  handleDelete: () => {},
  handleLink: () => {},
  canModify: true
};

export default ArticleOptions;
