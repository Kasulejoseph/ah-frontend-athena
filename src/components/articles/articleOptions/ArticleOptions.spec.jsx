import React from "react";
import { shallow } from "enzyme";
import ArticleOptions from "./ArticleOptions";
import RoundButton from "../../RoundButton/RoundButton";

describe("<ArticleOptions />", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<ArticleOptions />);
  });

  it("matches snapshort", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should an <RoundButton> element", () => {
    expect(wrapper.find(RoundButton)).toHaveLength(2);
  });
});
