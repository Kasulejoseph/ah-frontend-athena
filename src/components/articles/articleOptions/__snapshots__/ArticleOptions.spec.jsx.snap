// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArticleOptions /> matches snapshort 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ArticleOptions
    canModify={true}
    handleDelete={[Function]}
    handleLink={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <RoundButton
          background=""
          hoverClass="hover"
          icon="edit"
          label="Edit Article"
          onClick={[Function]}
        />,
        <RoundButton
          background=""
          hoverClass="hover"
          icon="trash"
          label="Delete Article"
          onClick={[Function]}
        />,
      ],
      "className": "d-flex flex-colomn",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "background": "",
          "hoverClass": "hover",
          "icon": "edit",
          "label": "Edit Article",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "background": "",
          "hoverClass": "hover",
          "icon": "trash",
          "label": "Delete Article",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <RoundButton
            background=""
            hoverClass="hover"
            icon="edit"
            label="Edit Article"
            onClick={[Function]}
          />,
          <RoundButton
            background=""
            hoverClass="hover"
            icon="trash"
            label="Delete Article"
            onClick={[Function]}
          />,
        ],
        "className": "d-flex flex-colomn",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "background": "",
            "hoverClass": "hover",
            "icon": "edit",
            "label": "Edit Article",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "background": "",
            "hoverClass": "hover",
            "icon": "trash",
            "label": "Delete Article",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
